scrapy/crawler.py

```diff
@@ -113,20 +113,27 @@ def _apply_settings(self) -> None:
 
         reactor_class: str = self.settings["TWISTED_REACTOR"]
         event_loop: str = self.settings["ASYNCIO_EVENT_LOOP"]
+        logged_reactor_info: bool = False
+
         if self._init_reactor:
             # this needs to be done after the spider settings are merged,
             # but before something imports twisted.internet.reactor
             if reactor_class:
                 install_reactor(reactor_class, event_loop)
             else:
                 from twisted.internet import reactor  # noqa: F401
-            log_reactor_info()
+
+            if not logged_reactor_info:
+                log_reactor_info()
+                logged_reactor_info = True
+
         if reactor_class:
             verify_installed_reactor(reactor_class)
             if is_asyncio_reactor_installed() and event_loop:
                 verify_installed_asyncio_event_loop(event_loop)
 
-            log_reactor_info()
```

Comment:
What if we only log it here, but remove 1 level of indentation?

Reply:
There are cases where neither of these two blocks is executed I think.

Reply:
Since this only logs information about the reactor being used, I wonder if it makes sense to only log in when those conditions are meant. Although maybe it does, maybe logging it always could trigger multiple logging because this function is called multiple times, but only once with the conditions for the logging.

To be safe, we could check the conditions, both, at this point:

```
if a and b:
    log()
```